graphrag_sdk/steps/extract_data_step.py

```
@@ -90,7 +90,12 @@ def run(self, instructions: str = None):
                 tasks.append(task)
 
             # Wait for all tasks to complete
-            wait(tasks)
+            done, _ = wait(tasks)  # Get completed tasks
+            for task in done:
+                try:
+                    task.result()  # Re-raise any exceptions from _process_source
+                except Exception as e:
+                    raise ValueError(f"Error in task: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)